utils.ts

```diff
@@ -0,0 +1,51 @@
+import { TestProvider, TestAccountSigningKey, Provider, Signer } from '@acala-network/bodhi';
+import { WsProvider, Keyring } from '@polkadot/api';
+import { ApiOptions, KeyringPair } from '@polkadot/api/types';
+import { createTestPairs } from '@polkadot/keyring/testingPairs';
+
+const DEFAULT_URL = 'ws://127.0.0.1:9944';
+
+export const getTestProvider = (urlOverwrite?: string, opts?: ApiOptions): TestProvider => {
+  const url = urlOverwrite || process.env.ENDPOINT_URL || DEFAULT_URL;
+
+  const provider = new TestProvider({
+    provider: new WsProvider(url),
+    ...opts
+  });
+
+  console.log(`test provider connected to ${url}`);
+
+  return provider;
+};
+
+export const setup = async (urlOverwrite?: string) => {
```

Comment:
this method is not used now, but is used by old waffle examples, so might be useful in the future for new waffle tutorials